Samples/OpenQasmReader.Tests/ParseApplicationTest.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Xunit;
+
+namespace Microsoft.Quantum.Samples.OpenQasmReader.Tests
+{
+    public class ParseApplicationTest
+    {
+        [Fact]
+        public void ParseHeaderTest()
+        {
+            var input = "OPENQASM 2.0;";
+            using (var stream = new StringReader(input))
+            {
+                var enumerator = Parser.Tokenizer(stream).GetEnumerator();
+
+                var cRegs = new Dictionary<string, int>();
+                var qRegs = new Dictionary<string, int>();
+                var inside = new StringBuilder();
+                var outside = new StringBuilder();
+                var conventionalMeasured = new List<string>();
```

Comment:
I'm not sure I understand the role of this variable, sorry? Is this used to record which classical registers hold the results of measurements?

Reply:
Indeed. Its to be able to register which need to be returned in the method.